libtransact/src/context/error.rs

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 Bitwise IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+use std::error::Error as StdError;
+
+use crate::receipts::TransactionReceiptBuilderError;
+use crate::state::error::StateReadError;
+
+#[derive(Debug)]
+pub enum ContextManagerError {
+    MissingContextError(String),
+    TransactionReceiptBuilderError(TransactionReceiptBuilderError),
+    StateReadError(StateReadError),
```

Comment:
You should implement `source` for these errors: https://doc.rust-lang.org/std/error/trait.Error.html#method.source

Reply:
Addressed this comment